qt-models/divetripmodel.cpp

```diff
@@ -683,8 +689,35 @@ dive *DiveTripModelTree::diveOrNull(const QModelIndex &index) const
 	return tripOrDive(index).dive;
 }
 
+bool DiveTripModelTree::setShown(const QModelIndex &idx, bool shown)
+{
+	if (!idx.isValid())
+		return false;
+
+	QModelIndex parent = idx.parent();
+	if (!parent.isValid()) {
+		// An invalid parent means that we're at the top-level
+		Item &item = items[idx.row()];
+		item.shown = shown;
+		if (item.d_or_t.dive)
+			filter_dive(item.d_or_t.dive, shown);
+	} else {
+		filter_dive(items[parent.row()].dives[idx.row()], shown);
+	}
+
+	return true;
```

Comment:
read through this function a couple of times... I'm still not entirely sure what it does, conceptually. I understand the code, but... 
Maybe I need to stare at all this a little longer, but I'll get on an airplane in a few minutes...

Reply:
Added a few comments and split a line in two. That's hopefully more clear now. Maybe we should cache the flag in the divetrip core-structure, but that can be done later. At the moment it's most important to fix that annoying crash.

Reply:
WiFi on the flight... the crash is fixed. I had two different ways to crash things (and one other situation where undo lost the selection). All of them work reliably now. Well done.